src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -329,7 +329,10 @@ void Run (DirectoryAssemblyResolver res, bool useMarshalMethods)
 					Log.LogCodedError ("XA4215", Properties.Resources.XA4215_Details, kvp.Key, typeName);
 			}
 
-			// NOTE: $(SupportedOSPlatformVersion) will potentially be 21.0
+			// NOTE: $(SupportedOSPlatformVersion) will potentially be 21.0, but could also be an int
+			if (SupportedOSPlatformVersion.IndexOf ('.') == -1) {
+				SupportedOSPlatformVersion += ".0";
+			}
```

Comment:
This logic is in this PR a couple times. Would we put a helper method in `MonoAndroidHelper.cs` to share the logic between multiple tasks?

Reply:
Good point, updated the tasks and tests.